File: practice/src/talker.cpp
#include "ros/ros.h"
#include "std_msgs/Int64.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "talker");
    
    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<std_msgs::Int64>("counter", 10);
    ros::Rate loop_rate(1);
    
    int number = 0;
    std_msgs::Int64 msg;
    
    while(ros::ok())
    {
        msg.data = number;
        // ROS_INFO("%d", msg.data);
        pub.publish(msg);
        
        number++;
        loop_rate.sleep();
    }
    return 0;
}

File: practice/src/talker.py
#!/usr/bin/env python3 
#coding:utf-8

import rospy
from std_msgs.msg import Int64

def publisher():
    rospy.init_node('talker')
    pub = rospy.Publisher('counter', Int64, queue_size = 10)
    count = 0
    while not rospy.is_shutdown():
            pub.publish(count)
            rospy.sleep(2)
            count+=1
    
if __name__=='__main__':
    try:
        publisher()
    except rospy.ROSInterruptException:
        pass